From 313c0343251dbc2d852e579fc6026b1980535ecb Mon Sep 17 00:00:00 2001
From: Sreenidhi B R <sreenira@broadcom.com>
Date: Tue, 17 Sep 2013 03:50:03 -0700
Subject: pktmem: handle size greater than 4GB.

also, use PAGE_SIZE macro instead of numerical value.
[Based on SDK 3.2]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
Signed-off-by: Nam Ninh <nam.ninh@windriver.com>

diff --git a/drivers/misc/netlogic/pkt_pool_mem/pkt_pool_mem.c b/drivers/misc/netlogic/pkt_pool_mem/pkt_pool_mem.c
index dd3676e..4502e80 100644
--- a/drivers/misc/netlogic/pkt_pool_mem/pkt_pool_mem.c
+++ b/drivers/misc/netlogic/pkt_pool_mem/pkt_pool_mem.c
@@ -19,7 +19,7 @@ typedef struct memory_pool
 {
 	struct memory_pool * next;
 	unsigned long long start;
-	uint32_t size;
+	unsigned long long size;
 	char shr_id[64];
 	int in_use;
 } memory_pool_t;
@@ -36,7 +36,7 @@ static int print_memory(brcm_devmem_ioctl_t * param)
 	printk("pktmem: printing for node %d.\n", param->node_id);
 	while (mem != NULL)
 	{
-		printk("pktmem: addr 0x%llx, size 0x%x, shr_id %s, in_use %d.\n",
+		printk("pktmem: addr 0x%llx, size 0x%llx, shr_id %s, in_use %d.\n",
 				mem->start, mem->size, mem->shr_id, mem->in_use);
 		mem = mem->next;
 	}
@@ -81,8 +81,8 @@ static int alloc_memory (brcm_devmem_ioctl_t * param)
 	/*
 	 * Align the packet allocation to 64K page size.
 	 */
-	if((align_len = param->size % 65536) != 0)
-		param->size += (65536 - align_len) ;
+	if((align_len = param->size % PAGE_SIZE) != 0)
+		param->size += (PAGE_SIZE - align_len) ;
 
 
 	while (mem != NULL)
@@ -301,7 +301,7 @@ static int __init brcm_xlp_ppm_init (void)
 	/* print the pktmem array */
 	for (i=0; i<MAX_NODES; i++)
 	{
-		printk("pktmem: node %d, addr 0x%llx size 0x%x.\n",
+		printk("pktmem: node %d, addr 0x%llx size 0x%llx.\n",
 				i, mem_pool_data[i].start, mem_pool_data[i].size);
 	}
 
-- 
1.7.1

